lib/streamlit/elements/arrow_vega_lite.py

```diff
@@ -155,7 +155,10 @@ def marshall(
         raise ValueError("Vega-Lite charts require a non-empty spec dict.")
 
     if "autosize" not in spec:
-        spec["autosize"] = {"type": "fit", "contains": "padding"}
+        if "vconcat" in spec and use_container_width:
+            spec["autosize"] = {"type": "fit-x", "contains": "padding"}
```

Comment:
nit: maybe add a quick comment here explaining why `fit-x` is a better config for vconcat